src/dbnode/persist/fs/files.go

```diff
@@ -291,55 +292,128 @@ func timeAndIndexFromFileName(fname string, componentPosition int) (time.Time, i
 	return t, int(index), nil
 }
 
-type infoFileFn func(fname string, infoData []byte)
+type forEachInfoFileSelector struct {
+	fileSetType    persist.FileSetType
+	contentType    persist.FileSetContentType
+	filePathPrefix string
+	namespace      ident.ID
+	shard          uint32 // shard only applicable for data content type
+}
 
-func forEachInfoFile(filePathPrefix string, namespace ident.ID, shard uint32, readerBufferSize int, fn infoFileFn) {
+type infoFileFn func(fname string, id FileSetFileIdentifier, infoData []byte)
+
+func forEachInfoFile(
+	args forEachInfoFileSelector,
+	readerBufferSize int,
+	fn infoFileFn,
+) {
```

Comment:
should this propagate errors back?

Reply:
Yeah, so I decided to stick with how it's currently modeled in which it skips erroneous entries.  I can open an issue perhaps?